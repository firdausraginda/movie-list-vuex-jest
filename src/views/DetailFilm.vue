<template>
  <div class="detail-film">
    <img
      :src="dataFilm.imgFilm"
      alt=""
    >
    <h1>{{dataFilm.judulFilm}}</h1>
    <p>{{dataFilm.deskripsiFilm}}</p>
  </div>
</template>

<script>
export default {
  data() {
    return {
      dataFilm: null
    };
  },
  created() {
    this.getDataFilmDetail();
  },
  methods: {
    getDataFilmDetail() {
      this.$store.state.daftarFilm.forEach(element => {
        if (element.idFilm == this.$route.params.id) {
          this.dataFilm = element;
        }
      });
    }
  }
};
</script>

<style lang="scss" scoped>
.detail-film {
  padding: 0 150px;
  img {
    border-radius: 5px;
    height: 200px;
    object-fit: cover;
    margin-bottom: 20px;
  }
}
</style>