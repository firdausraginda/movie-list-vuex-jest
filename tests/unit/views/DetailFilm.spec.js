import { shallowMount } from '@vue/test-utils'
import DetailFilm from '../../../src/views/DetailFilm.vue'

describe('DetailFilm.spec.js', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(DetailFilm, {
            data() {
                return {
                    dataFilm: [{ idFilm: 'gans', choosen: false, imgFilm: 'https://upload.wikimedia.org/wikipedia/en/thumb/5/53/Black_mirror_bandersnatch_poster.jpg/220px-Black_mirror_bandersnatch_poster.jpg', judulFilm: 'Black Mirror: Bandersnatch', deskripsiFilm: 'Black Mirror: Bandersnatch adalah film interaktif tahun 2018 dalam seri antologi fiksi ilmiah Black Mirror. Film ini ditulis oleh pencipta serinya, Charlie Brooker, dan disutradarai oleh David Slade. Netflix merilisnya sebagai film terpisah pada tanggal 28 Desember 2018.' }]
                }
            },
            methods: {
                getDataFilmDetail() {
                    return [{ idFilm: 'gans', choosen: false, imgFilm: 'https://upload.wikimedia.org/wikipedia/en/thumb/5/53/Black_mirror_bandersnatch_poster.jpg/220px-Black_mirror_bandersnatch_poster.jpg', judulFilm: 'Black Mirror: Bandersnatch', deskripsiFilm: 'Black Mirror: Bandersnatch adalah film interaktif tahun 2018 dalam seri antologi fiksi ilmiah Black Mirror. Film ini ditulis oleh pencipta serinya, Charlie Brooker, dan disutradarai oleh David Slade. Netflix merilisnya sebagai film terpisah pada tanggal 28 Desember 2018.' }]
                }
            }
        })
    })

    test('is DetailFilm component', () => {
        expect(wrapper.is(DetailFilm)).toBe(true)
        expect(wrapper.is('.detail-film')).toBe(true)
    })

    test('has "dataFilm" property and not null', () => {
        expect(wrapper.vm.dataFilm).not.toBeNull()
        expect(wrapper.vm.dataFilm[0].idFilm).toBe('gans')
        expect(typeof wrapper.vm.dataFilm).toBe('object')
        expect(wrapper.vm.dataFilm[0].imgFilm).toBe('https://upload.wikimedia.org/wikipedia/en/thumb/5/53/Black_mirror_bandersnatch_poster.jpg/220px-Black_mirror_bandersnatch_poster.jpg')
        expect(wrapper.vm.dataFilm[0].idFilm).toBe('gans')
        expect(wrapper.vm.dataFilm[0].judulFilm).toBe('Black Mirror: Bandersnatch')
    })

    test('has the expected html structure', () => {
        expect(wrapper.element).toMatchSnapshot()
    })

    test('"getDataFilmDetail" is a funciton and not null', () => {
        expect(typeof wrapper.vm.getDataFilmDetail).toBe('function')
        expect(wrapper.vm.getDataFilmDetail).not.toBeNull()
    })

})