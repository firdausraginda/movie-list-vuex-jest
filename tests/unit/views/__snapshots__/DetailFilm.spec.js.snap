// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailFilm.spec.js has the expected html structure 1`] = `
<div
  class="detail-film"
>
  <img
    alt=""
  />
   
  <h1>
    
  </h1>
   
  <p>
    
  </p>
</div>
`;
